### PR TITLE
Fix Rhode Island property tax calculation

### LINKED ISSUE #3670: Fix Rhode Island property tax calculation
In the 0.631.0 version, the RI property tax credit is not always correctly calculated.   In the test case below, both members of the childless couple are age 66 and there income is well above $30,000 (the value of `ri_property_tax_household_income` is $65,030).  Yet PEUS generates a $500 property tax credit (while TAXSIM35 considers the couple ineligible for the credit).

```
- name: Tax unit with taxsimid 99830 in g21.its.csv and g21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        age: 66
        employment_income: 18010
        taxable_interest_income: 5505
        taxable_private_pension_income: 500
        social_security_retirement: 4500
        rent: 22000
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        age: 66
        employment_income: 26010
        taxable_interest_income: 5505
        taxable_private_pension_income: 500
        social_security_retirement: 4500
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2]
        premium_tax_credit: 0  # not in TAXSIM35
        local_income_tax: 0  # not in TAXSIM35
        state_sales_tax: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2]
        state_fips: 44  # RI
  output:  # expected results from patched TAXSIM35 2024-01-14 version
    ri_income_tax: 1066.13
```

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3670

### CODE REVIEW

**Review by MaxGhenis — COMMENTED** [review]
Could you summarize what caused the issue and how this resolves it?

**Comment by PavelMakarchuk** [comment]
> Could you summarize what caused the issue and how this resolves it?

The issue was caused mainly by the missing check for household income eligibility. While we have used a 0% rate in the parameter files when calculating the actual rate of income which is used for the credit computation, it did not actually result in the credit being 0 
By adding this check to the eligibility file, we circumvent that
```
        household_income = tax_unit("ri_property_tax_household_income", period)
        household_income_eligible = (
            household_income <= p.rate.one_person.thresholds[0]
        )
```
Other changes, like the consolidation of the household income, are me consolidating the logic to the legal code as opposed to the tax form

**Review by MaxGhenis — APPROVED** [review]
